### DISCUSSION TITLE
Is there a complete example of auth with SSR (or more in depth docs)?

### DISCUSSION THREAD

**Discussion by blacksandsolutions** [supabase/supabase#19616] [Questions]
I have auth setup using the new SSR package.
I am now tying to prevent un-authenticated users from accessing protected routes.
However all examples I've found are for the old auth-helpers, which the docs suggest should not be used anymore.

Furthermore I see the new package is actually still in beta - should I ignore the docs and use the old auth helpers instead?
My application uses the NextJS 14 app router.

UPDATE:

I have added the following to `middelware` 

```ts

    const supabase = createServerClient(...)

    const auth = await supabase.auth.getSession()

    const isAuthenticated = auth.data.session != null

    if (!isAuthenticated && !publicRoutes.includes(request.nextUrl.pathname)) {
        const absoluteURL = new URL('/login', request.nextUrl.origin)        
        return NextResponse.redirect(absoluteURL.toString())
    } else {
        return response
    }
```

However, while the if block does run, `NextResponse.redirect` does not redirect to `/login`, nothing happens.

UPDATE2:

Turns out I had the code in the wrong `middleware` file. 
Moved to the correct one and it now redirects to login page, but I get a bunch of errors trying to render the page (render worked find before this change)

<img width="1557" alt="Screenshot 2023-12-12 at 5 45 18 PM" src="https://github.com/supabase/supabase/assets/58924459/d4aa3410-1a8e-4727-85d2-dd214a09f264">

The only change is to this file
![Screenshot 2023-12-12 at 5 46 43 PM](https://github.com/supabase/supabase/assets/58924459/93fbf42b-394b-41e9-b604-501aa8d19e5d)

UPDATE3:

So I was accidently intercepting some ` /_next/*` requests and redirecting those to `/login`.

This was resolved by adding this to the middleware file

```ts
export const config = {
  matcher: [
    '/((?!_next|api/auth).*)(.+)'
  ],
}
```

```

**Comment by zhamid-src** [comment]
Did you get the password reset flow working with SSR? I've wasted 2 days on it and am ready to give up on supabase.

**Reply by blacksandsolutions** [reply]
Not tried yet. TBH I think my issues were more NextJS and SSR related, rather than Supabase itself. 
Might try reset password this weekend if I get time.

**Reply by cubdio** [reply]
Hi @blacksandsolutions, I'm having a similar issue and wondered if you might have a moment to take a look at my code and see if you can identify the issue? I would be extremely grateful.

I've been banging my head against a wall for days trying to figure this out. 

After my user successfully authenticates, they get redirected to the homepage no matter what, I've tried everything I can possible think of to get them to redirect to my /dashboard page. But I'm sure I must be missing something obvious at this point.

Repository: https://github.com/cubdio/cubd-app

Note: I'm also using Next.js 14

**Reply by charislam** [reply]
Hi @cubdio, maybe you've already done this, but have you double-checked that your redirect URL `<YOUR_DOMAIN_HERE>/dashboard` is on your allow list? https://supabase.com/docs/guides/auth/concepts/redirect-urls

**Reply by zhamid-src** [reply]
I ended up giving up on supabase. Sad because it had potential.

**Comment by blacksandsolutions** [comment]
Hi,

I've not worked on this project much lately as another priority came in.
However I had a quick look and I appear to be using a server side action
for the signin


'use server'

import { cookies } from 'next/headers'
import { createClient } from '@/utils/supabase/server'
import { redirect } from 'next/navigation'

export type FormState = {
type: 'success' | 'error'
errors: any
}

export const signIn = async (
prevState: string | undefined,
formData: FormData,
): Promise<never | FormState> => {
const email = formData.get('email') as string
const password = formData.get('password') as string
const cookieStore = cookies()
const supabase = createClient(cookieStore)

const { error } = await supabase.auth.signInWithPassword({
email,
password,
})

if (error) {
console.log(error)
//return redirect('/login?message=Could not authenticate user')
return {
type: 'error' as const,
errors: error.message,
}
}

return redirect('/dashboard')
}


On Sat, 30 Dec 2023 at 07:08, cubdio ***@***.***> wrote:

> Hi @blacksandsolutions <https://github.com/blacksandsolutions>, I'm
> having a similar issue and wondered if you might have a moment to take a
> look at my code and see if you can identify the issue?
>
> I've been banging my head against a wall for days trying to figure this
> out.
>
> After my user successfully authenticates, they get redirected to the
> homepage no matter what, I've tried everything I can possible think of to
> get them to redirect to my /dashboard page. But I'm sure I must be missing
> something obvious at this point.
>
> Repository: https://github.com/cubdio/cubd-app
>
> Note: I'm also using Next.js 14
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/supabase/discussions/19616#discussioncomment-7974817>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AOBR3K4M6XTHJMKZ2SBR7LLYL4BLNAVCNFSM6AAAAABAQYEOZ2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM3TSNZUHAYTO>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Rich Field

*Black Sand Solutions*
www.blacksandsolutions.co
021 02244767
